dev/reloadable-server/src/main/scala/play/core/server/LagomReloadableDevServerStart.scala

```diff
@@ -142,88 +142,83 @@ object LagomReloadableDevServerStart {
               // and https://github.com/playframework/playframework/pull/7644
               //
               // we should reconsider if still need to have our own reloadable server
-              implicit val ec = ExecutionContext.global
-              Await.result(scala.concurrent.Future {
```

Comment:
The main change was to take out the `Future` spawn + `Await`. It's not obvious what the point of that was, but I believe that blocking on the global execution context (without signaling it using [`blocking`](https://www.scala-lang.org/api/current/scala/concurrent/index.html#blocking[T](body:=%3ET):T)) was causing the global execution context to run out of threads.

By default, the global execution context has max threads = #CPUs, making this more likely to occur in constrained environments such as Travis CI workers.

Reply:
Note that, prior to the changes in #1405, this [ran on the Play internal execution context](https://github.com/lagom/lagom/pull/1405/files#diff-b0570315055c9029853b183c2943bf7aL139), which has been removed in Play 2.7.